### PR TITLE
Tunnel increase bytes fix

### LINKED ISSUE #2283: Invalid call to increase_bytes_sent() in routing.py
https://github.com/Tribler/tribler/blob/devel/Tribler/community/tunnel/routing.py#L104

Here increase_bytes_sent() is called with the circuit_id (of type long) as the first argument.
However, increase_bytes_sent() only does something when called with a Circuit, RelayRoute or TunnelExitSocket object. This leads to several useless calls per second.

We should either remove this statement or fix it. Additionally, increase_bytes_sent() should probably explicitly fail when called with an improper value.

### PR DESCRIPTION
**Pull request by pimveldhuisen**
Fixes #2283

### CODE REVIEW

**Comment by pimveldhuisen** [comment]
```
Test raised 3 unhandled Twisted exceptions, last one was: Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/Utilities/twisted_thread.py", line 55, in _reactor_runner
    reactor.run(installSignalHandlers=False)
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1194, in run
    self.mainLoop()
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1203, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 798, in runUntilCurrent
    f(*a, **kw)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/dispersy/endpoint.py", line 250, in dispersythread_data_came_in
    u"standalone_ep")
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/dispersy/dispersy.py", line 1464, in on_incoming_packets
    (str(packet[0]), repr(packet[1])) for packet in packets]
exceptions.AssertionError: [('<Candidate 79.158.21.215:43892>', "'A\\x00%\\xa9$\\x80\\xc5\\xe4\\x00\\x00\\x00\\x00\\x00\\x10\\x00\\x00(\\xd3\\x00\\x00'")]
```

**Comment by pimveldhuisen** [comment]
retest this please

File: Tribler/community/tunnel/tunnel_community.py [hunk 1]
```diff
@@ -1374,6 +1374,8 @@ def increase_bytes_sent(self, obj, num_bytes):
             self.stats['bytes_exit'] += num_bytes
             _barter_statistics.dict_inc_bartercast(BartercastStatisticTypes.TUNNELS_EXIT_BYTES_SENT, "%s:%s" %
                                                    (obj.sock_addr[0], obj.sock_addr[1]), num_bytes)
+        else:
+            raise TypeError
```
Comment (whirm) [inline Tribler/community/tunnel/tunnel_community.py]: Please, make the error more informative to ease debugging

**Comment by whirm** [comment]
Remember to have the PRs reviewed by a peer before marking them as read. 

Please read [this](https://whirm.eu/posts/triblers-branching-model/).

**Comment by pimveldhuisen** [comment]
@whirm Nice guide! I will do both things.

**Comment by pimveldhuisen** [comment]
@lfdversluis or @devos50 Could you review this PR?

**Comment by pimveldhuisen** [comment]
I added some tests for the error branches

**Comment by pimveldhuisen** [comment]
````
Build timed out (after 15 minutes). Marking the build as failed.
````

**Comment by pimveldhuisen** [comment]
retest this please

**Comment by whirm** [comment]
The last comment from @devos50 hasn't been addressed yet?

**Comment by pimveldhuisen** [comment]
@whirm Hmm, weird, I made the tests in e4bc8e9fcba4c0020349f161547db5380690c21d to cover them, but it doesn't seem to work?

**Comment by lfdversluis** [comment]
@pimveldhuisen place a print in line 1399 and see if it fires :)

**Comment by whirm** [comment]
retest this please

**Comment by whirm** [comment]
tunnel tests stuck again

**Comment by whirm** [comment]
retest this please

File: Tribler/Test/Community/Tunnel/test_tunnelcommunity.py [hunk 2]
```diff
@@ -120,3 +120,9 @@ def test_send_to_ip_deferred(self):
         data = "ffffffff".decode("HEX") + "1" * 25
         exit_tunnel.sendto(data, ("localhost", -1))
         return exit_tunnel.close()
+
+    def test_increase_bytes_sent_error_branch(self):
```
Comment (whirm) [inline Tribler/Test/Community/Tunnel/test_tunnelcommunity.py]: Shouldn't this be called on the reactor thread?

**Comment by pimveldhuisen** [comment]
Wow, I figured out why the line wasn't covered: I called AssertRaises with the wrong signature, which resulted in a TypeError, but since I happend to check for a TypeError, the test actually succeeded :)

File: Tribler/Test/Community/Tunnel/test_tunnelcommunity.py (hunk 2 above)
Comment (pimveldhuisen) [inline Tribler/Test/Community/Tunnel/test_tunnelcommunity.py]: @whirm This only checks if the error is raised, so I think it should not matter?

**Comment by pimveldhuisen** [comment]
On Windows 32, 
```
Tribler.Test.Core.Modules.RestApi.test_channels_endpoints.TestChannelsEndpoint.test_get_subscribed_channels_one_subscription
```
failed

**Comment by pimveldhuisen** [comment]
retest this please

**Comment by pimveldhuisen** [comment]
On Windows 64, 
```
Tribler.Test.Core.Modules.RestApi.test_channels_endpoints.TestChannelsEndpoint.test_get_subscribed_channels_one_subscription
```
failed

**Comment by pimveldhuisen** [comment]
retest this please

**Comment by synctext** [comment]
@whirm @pimveldhuisen spend an hour hunting hints in debug info: https://github.com/Tribler/tribler/issues/2283#issuecomment-228589016

**Comment by whirm** [comment]
Squash it!

**Comment by pimveldhuisen** [comment]
retest this please

**Comment by pimveldhuisen** [comment]
Yay, 7/7. Merge?

**Pull request merged**